cshared.go

```diff
@@ -438,6 +445,59 @@ func FLBPluginFlush(data unsafe.Pointer, clength C.int, ctag *C.char) int {
 	}
 
 	in := C.GoBytes(data, clength)
+	tag := C.GoString(ctag)
+
+	return pluginFlush(tag, in)
+}
+
+func decodeEntry(tag string, entry []any) (*Message, error) {
+	var t time.Time
+
+	slice := reflect.ValueOf(entry)
```

Comment:
I know you just moved the code around and changed downbelow the record type, but is there a need to use reflection here? I don't think so...

```go
if len(entry) != 2 { ... }

var t time.Time
switch t := entry[0].(type) { ... }

data, ok := entry[1](map[string]any])
...
```

Reply:
But the msgpack decoder will already handle the time and use `bigEndianTime` type. I think we can just go with type assertion instead of reflection.

Reply:
Ah, it's not effective. Because once decoded for metadata part which includes timestamp additional metadata, it's just represented with wrapped up as msgpack map not array.

Reply:
I modified the code in `decodeEntry` to forego the use of `reflect`.